Prefer jakarta apis over older javax apis.

pkwarren opened the pull request:
###### Problem:
When creating a shaded jar based on dropwizard APIs, warnings are shown
during shading about duplicate classes on the classpath.

This occurs because there is a mix of javax.* and jakarta.* modules on the
runtime classpath which ship the same (but conflicting) versions of classes.

###### Solution:
Update to use the latest jakarta apis in all dropwizard modules instead
of a mix of jakarta and javax.* ones (which ship the same classes). This
should make the APIs on the classpath consistent when creating a shaded
application jar and also avoid warnings about duplicate classes during
shading.

To validate that jakarta apis are used, a Maven profile `jakarta-apis` has
been added to the build. To verify jakarta APIs are used, run the build 
with the argument `-P jakarta-apis`.

###### Result:
Should have no impact to downstream applications and avoid shading
warnings about duplicate classes.

pkwarren commented on dropwizard-auth/src/main/java/io/dropwizard/auth/CachingAuthorizer.java:
```diff
@@ -21,45 +21,6 @@
 
 import static com.codahale.metrics.MetricRegistry.name;
 
-class AuthorizationContext<P extends Principal> {
```
This was a top-level class defined in a different java file for some reason. I've extracted it to the top-level package to make it easier to find where this code is found.

It makes sense to me to either package as an inner class or a top-level class in a separate .java file, but not in the same .java file as another class.

pkwarren commented:
Dependency diffs:

```diff
--- ./dropwizard-assets/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-assets/new-deps.txt	2019-12-17 11:25:13.000000000 -0600
@@ -49,14 +49,12 @@
    io.dropwizard:dropwizard-servlets:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.servlets (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-auth/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-auth/new-deps.txt	2019-12-17 11:25:12.000000000 -0600
@@ -49,15 +49,12 @@
    io.dropwizard:dropwizard-servlets:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.servlets (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:test -- module junit (auto)
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
--- ./dropwizard-benchmarks/old-deps.txt	2019-12-17 11:25:00.000000000 -0600
+++ ./dropwizard-benchmarks/new-deps.txt	2019-12-17 11:25:14.000000000 -0600
@@ -33,15 +33,12 @@
    io.dropwizard:dropwizard-logging:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.logging (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
    jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:compile -- module junit (auto)
    net.sf.jopt-simple:jopt-simple:jar:4.6:compile -- module jopt.simple (auto)
--- ./dropwizard-client/old-deps.txt	2019-12-17 11:24:58.000000000 -0600
+++ ./dropwizard-client/new-deps.txt	2019-12-17 11:25:12.000000000 -0600
@@ -53,15 +53,12 @@
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
    io.reactivex.rxjava2:rxjava:jar:2.0.4:test -- module rxjava (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:test -- module junit (auto)
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
--- ./dropwizard-core/old-deps.txt	2019-12-17 11:24:57.000000000 -0600
+++ ./dropwizard-core/new-deps.txt	2019-12-17 11:25:11.000000000 -0600
@@ -48,15 +48,12 @@
    io.dropwizard:dropwizard-servlets:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.servlets (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:test -- module junit (auto)
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
--- ./dropwizard-db/old-deps.txt	2019-12-17 11:24:58.000000000 -0600
+++ ./dropwizard-db/new-deps.txt	2019-12-17 11:25:12.000000000 -0600
@@ -50,14 +50,12 @@
    io.dropwizard:dropwizard-servlets:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.servlets (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-e2e/old-deps.txt	2019-12-17 11:25:00.000000000 -0600
+++ ./dropwizard-e2e/new-deps.txt	2019-12-17 11:25:14.000000000 -0600
@@ -57,15 +57,12 @@
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
    io.dropwizard:dropwizard-views-mustache:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.views.mustache (auto)
    io.dropwizard:dropwizard-views:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.views (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
    jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:compile -- module junit (auto)
    net.sourceforge.argparse4j:argparse4j:jar:0.8.1:compile -- module argparse4j (auto)
--- ./dropwizard-example/old-deps.txt	2019-12-17 11:25:00.000000000 -0600
+++ ./dropwizard-example/new-deps.txt	2019-12-17 11:25:13.000000000 -0600
@@ -69,17 +69,14 @@
    io.dropwizard:dropwizard-views-freemarker:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.views.freemarker (auto)
    io.dropwizard:dropwizard-views-mustache:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.views.mustache (auto)
    io.dropwizard:dropwizard-views:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.views (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    javax.persistence:javax.persistence-api:jar:2.2:compile -- module java.persistence [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
    javax.transaction:javax.transaction-api:jar:1.3:compile -- module java.transaction [auto]
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:compile -- module net.bytebuddy
--- ./dropwizard-forms/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-forms/new-deps.txt	2019-12-17 11:25:12.000000000 -0600
@@ -49,14 +49,12 @@
    io.dropwizard:dropwizard-servlets:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.servlets (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.sourceforge.argparse4j:argparse4j:jar:0.8.1:compile -- module argparse4j (auto)
    org.apache.commons:commons-lang3:jar:3.9:compile -- module org.apache.commons.lang3 [auto]
--- ./dropwizard-hibernate/old-deps.txt	2019-12-17 11:24:58.000000000 -0600
+++ ./dropwizard-hibernate/new-deps.txt	2019-12-17 11:25:12.000000000 -0600
@@ -56,17 +56,14 @@
    io.dropwizard:dropwizard-testing:jar:2.0.1-SNAPSHOT:test -- module dropwizard.testing (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    javax.persistence:javax.persistence-api:jar:2.2:compile -- module java.persistence [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
    javax.transaction:javax.transaction-api:jar:1.3:compile -- module java.transaction [auto]
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:test -- module junit (auto)
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
--- ./dropwizard-http2/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-http2/new-deps.txt	2019-12-17 11:25:13.000000000 -0600
@@ -50,15 +50,12 @@
    io.dropwizard:dropwizard-testing:jar:2.0.1-SNAPSHOT:test -- module dropwizard.testing (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:test -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:test -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:test -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:test -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:test -- module junit (auto)
    net.sourceforge.argparse4j:argparse4j:jar:0.8.1:test -- module argparse4j (auto)
--- ./dropwizard-jdbi3/old-deps.txt	2019-12-17 11:24:58.000000000 -0600
+++ ./dropwizard-jdbi3/new-deps.txt	2019-12-17 11:25:12.000000000 -0600
@@ -53,14 +53,12 @@
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
    io.leangen.geantyref:geantyref:jar:1.3.7:compile -- module geantyref (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-jersey/old-deps.txt	2019-12-17 11:24:57.000000000 -0600
+++ ./dropwizard-jersey/new-deps.txt	2019-12-17 11:25:11.000000000 -0600
@@ -32,15 +32,12 @@
    io.dropwizard:dropwizard-logging:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.logging (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:test -- module junit (auto)
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
--- ./dropwizard-jetty/old-deps.txt	2019-12-17 11:24:57.000000000 -0600
+++ ./dropwizard-jetty/new-deps.txt	2019-12-17 11:25:11.000000000 -0600
@@ -29,8 +29,8 @@
    io.dropwizard:dropwizard-logging:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.logging (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:provided -- module javax.servlet.api (auto)
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-json-logging/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-json-logging/new-deps.txt	2019-12-17 11:25:13.000000000 -0600
@@ -32,8 +32,8 @@
    io.dropwizard:dropwizard-request-logging:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.request.logging (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-lifecycle/old-deps.txt	2019-12-17 11:24:57.000000000 -0600
+++ ./dropwizard-lifecycle/new-deps.txt	2019-12-17 11:25:11.000000000 -0600
@@ -26,8 +26,8 @@
    io.dropwizard:dropwizard-logging:jar:2.0.1-SNAPSHOT:test -- module dropwizard.logging (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:test -- module dropwizard.validation (auto)
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:runtime -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:test -- module java.validation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-metrics-graphite/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-metrics-graphite/new-deps.txt	2019-12-17 11:25:13.000000000 -0600
@@ -27,8 +27,8 @@
    io.dropwizard:dropwizard-metrics:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.metrics (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:runtime -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-metrics/old-deps.txt	2019-12-17 11:24:57.000000000 -0600
+++ ./dropwizard-metrics/new-deps.txt	2019-12-17 11:25:11.000000000 -0600
@@ -29,8 +29,8 @@
    io.dropwizard:dropwizard-logging:jar:2.0.1-SNAPSHOT:test -- module dropwizard.logging (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:runtime -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-migrations/old-deps.txt	2019-12-17 11:24:58.000000000 -0600
+++ ./dropwizard-migrations/new-deps.txt	2019-12-17 11:25:12.000000000 -0600
@@ -55,14 +55,12 @@
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
    io.leangen.geantyref:geantyref:jar:1.3.7:test -- module geantyref (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-request-logging/old-deps.txt	2019-12-17 11:24:57.000000000 -0600
+++ ./dropwizard-request-logging/new-deps.txt	2019-12-17 11:25:11.000000000 -0600
@@ -31,8 +31,8 @@
    io.dropwizard:dropwizard-logging:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.logging (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-servlets/old-deps.txt	2019-12-17 11:24:57.000000000 -0600
+++ ./dropwizard-servlets/new-deps.txt	2019-12-17 11:25:11.000000000 -0600
@@ -12,7 +12,7 @@
    io.dropwizard.metrics:metrics-annotation:jar:4.1.2:compile -- module com.codahale.metrics.annotation [auto]
    io.dropwizard.metrics:metrics-core:jar:4.1.2:compile -- module com.codahale.metrics [auto]
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
-   javax.servlet:javax.servlet-api:jar:4.0.1:provided -- module javax.servlet.api (auto)
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
--- ./dropwizard-testing/old-deps.txt	2019-12-17 11:24:58.000000000 -0600
+++ ./dropwizard-testing/new-deps.txt	2019-12-17 11:25:11.000000000 -0600
@@ -53,16 +53,13 @@
    io.dropwizard:dropwizard-servlets:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.servlets (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
    jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    javax.persistence:javax.persistence-api:jar:2.2:provided -- module java.persistence [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:compile (optional)  -- module junit (auto)
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:compile (optional)  -- module net.bytebuddy.agent
--- ./dropwizard-views-freemarker/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-views-freemarker/new-deps.txt	2019-12-17 11:25:13.000000000 -0600
@@ -50,15 +50,12 @@
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
    io.dropwizard:dropwizard-views:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.views (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:test -- module junit (auto)
    net.sourceforge.argparse4j:argparse4j:jar:0.8.1:compile -- module argparse4j (auto)
--- ./dropwizard-views-mustache/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-views-mustache/new-deps.txt	2019-12-17 11:25:13.000000000 -0600
@@ -51,15 +51,12 @@
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
    io.dropwizard:dropwizard-views:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.views (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
-   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:test -- module java.servlet [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    junit:junit:jar:4.12:test -- module junit (auto)
    net.sourceforge.argparse4j:argparse4j:jar:0.8.1:compile -- module argparse4j (auto)
--- ./dropwizard-views/old-deps.txt	2019-12-17 11:24:59.000000000 -0600
+++ ./dropwizard-views/new-deps.txt	2019-12-17 11:25:13.000000000 -0600
@@ -49,14 +49,12 @@
    io.dropwizard:dropwizard-servlets:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.servlets (auto)
    io.dropwizard:dropwizard-util:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.util (auto)
    io.dropwizard:dropwizard-validation:jar:2.0.1-SNAPSHOT:compile -- module dropwizard.validation (auto)
-   jakarta.activation:jakarta.activation-api:jar:1.2.1:compile -- module jakarta.activation [auto]
+   jakarta.activation:jakarta.activation-api:jar:1.2.1:runtime -- module jakarta.activation [auto]
    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile -- module java.annotation [auto]
+   jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile -- module java.servlet [auto]
    jakarta.validation:jakarta.validation-api:jar:2.0.2:compile -- module java.validation [auto]
    jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile -- module java.ws.rs
-   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile -- module java.xml.bind
-   javax.activation:javax.activation-api:jar:1.2.0:runtime -- module java.activation [auto]
-   javax.servlet:javax.servlet-api:jar:4.0.1:compile -- module javax.servlet.api (auto)
-   javax.xml.bind:jaxb-api:jar:2.3.1:runtime -- module java.xml.bind
+   jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:runtime -- module java.xml.bind
    joda-time:joda-time:jar:2.10.5:compile -- module org.joda.time [auto]
    net.bytebuddy:byte-buddy-agent:jar:1.10.5:test -- module net.bytebuddy.agent
    net.bytebuddy:byte-buddy:jar:1.10.5:test -- module net.bytebuddy
```

mattnelson commented:
> To validate that jakarta apis are used, a Maven profile `jakarta-apis` has
> been added to the build. To verify jakarta APIs are used, run the build
> with the argument `-P jakarta-apis`.

This should be active by default and/or enabled on the CI builds.

pkwarren commented:
@mattnelson - Good point on CI builds. I wasn't sure if we want to enable the profile by default since the DropWizard 2.x upgrade notes mention that you can add dropwizard-dependencies as a parent pom and this might break existing projects.

pkwarren commented:
This should be ready for review again. I've added the jakarta-apis profile to CI builds and also fixed a flaky test which was showing up on some of the matrix builds by bumping the default Jersey client timeouts. A new version of liquibase-core also pulled in javax.xml.bind:jaxb-api as a transitive dependency so I've migrated that to the jakarta version in dropwizard-migrations.

jplock commented:
@pkwarren are there any places in the docs that we should update to include whatever runtime libraries should be required?

pkwarren commented:
@jplock - I don't think there is any impact to users - we used to pull in two versions of the same library and now only pull in one.

It might be worthwhile to mention in 2.x upgrade notes the old vs. new artifact coordinates for common libraries (like `javax.validation:validation-api` -> `jakarta.validation:jakarta.validation-api` or `javax.ws.rs:javax.ws.rs-api` -> `jakarta.ws.rs:jakarta.ws.rs-api`).

pkwarren commented:
@jplock - Will this fix be in 2.0.1? Any idea when it may be pulled into a release? Thanks!

jplock commented:
We should probably merge this into the 2.0.x release branch. Not sure when I’ll be able to do that unfortunately

marmelo commented:
While this change is not released I suggest to **carefully exclude** the conflicting `javax.*` dependencies:

```
<dependency>
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-core</artifactId>
    <exclusions>
        <exclusion>
            <groupId>javax.xml.bind</groupId>
            <artifactId>jaxb-api</artifactId>
        </exclusion>
        <exclusion>
            <groupId>javax.activation</groupId>
            <artifactId>javax.activation-api</artifactId>
        </exclusion>
    </exclusions> 
</dependency>
```
